### PR TITLE
Fix potential Null pointer dereference in regress_ws.c

### LINKED ISSUE #1604: Potential Null pointer dereference in regress_ws.c
1. In test/regress_ws.c: 354, calling [bufferevent_setcb](https://github.com/libevent/libevent/blob/master/test/regress_ws.c#L354) without checking the parameter 1 might cause null dereference.
2. In test/regress_ws.c: 372, calling [bufferevent_setcb](https://github.com/libevent/libevent/blob/master/test/regress_ws.c#L372) without checking the parameter 1 might cause null dereference.

### PR DESCRIPTION
**Pull request by icy17**
Fix #1604

### CODE REVIEW

**Comment by icy17** [comment]
I'm not sure if I should close `fd` before `tt_assert`...

**Comment by azat** [comment]
>I'm not sure if I should close fd before tt_assert...

Yes, it should be done under `end` label, and you need to initialize fd with `-1`

**Review by azat — CHANGES_REQUESTED** [review]
.

File: test/regress_ws.c [hunk 1]
```diff
@@ -388,4 +390,5 @@ http_ws_test(void *arg)
 end:
 	if (bev)
 		bufferevent_free(bev);
+	EVUTIL_CLOSESOCKET(fd);
```
Comment (azat) [inline test/regress_ws.c:393]: `if (fd != EVUTIL_INVALID_SOCKET)` (and use it for initilzation), and also let's add a check for `http`

Comment (icy17) [reply test/regress_ws.c:393]: I noticed that the `http` is not used here? I'm not quite sure where I'm supposed to add the check. 
Additionally, I should initialize fd to `EVUTIL_INVALID_SOCKET` and check if it is `EVUTIL_INVALID_SOCKET` before closing,  is that right?

Comment (azat) [reply test/regress_ws.c:393]: > I noticed that the http is not used here?

It is used it creates http server, see usage of `port`.

> I'm not quite sure where I'm supposed to add the check.

The `http` initialization may fail, so - `if (!http) goto err`

>Additionally, I should initialize fd to EVUTIL_INVALID_SOCKET and check if it is EVUTIL_INVALID_SOCKET before closing, is that right?

Yes

**Review by azat — CHANGES_REQUESTED** [review]
```
/home/runner/work/libevent/libevent/test/regress_ws.c:351:2: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
  351 |  struct evbuffer *out;
      |  ^~~~~~
```

File: test/regress_ws.c [hunk 2]
```diff
@@ -340,17 +340,20 @@ http_ws_test(void *arg)
...
 	struct basic_test_data *data = arg;
 	struct bufferevent *bev = NULL;
-	evutil_socket_t fd;
+	evutil_socket_t fd = EVUTIL_INVALID_SOCKET;
 	ev_uint16_t port = 0;
 	int ssl = 0;
 	struct evhttp *http = http_setup(&port, data->base, ssl);
+	if (!http) 
+		goto end;
 	struct evbuffer *out;
```
Comment (Coeur) [inline test/regress_ws.c:349]: It's C, not C++, so the order matters:
```suggestion
	struct evbuffer *out;
	if (!http) 
		goto end;
```

**Comment by Coeur** [comment]
There seems to be a lot of failures with:
>[warn] evbuffer_file_segment_materialize: mmap(5, 0, 0) failed: Bad file descriptor

Could those failures be related to those changes?

**Comment by azat** [comment]
Please resubmit

**Pull request closed without merging**